ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -410,18 +410,24 @@ let assert_usbs_available ~__context ~self ~host =
       ]))
     )
 
-(* Get SR-IOV Vifs by return a list of [(network1,(required_num1,PCI1));(network2,(required_num2,PCI2))....] 
+(* 1.To avoid redundant checks,if VF had been reserved, then it's no need to check the ability again.
```

Comment:
Is this true: if any VF has been reserved, all other VFs are supposed to be reserved also?

Reply:
Yes, I think so. Because at here the VF(s) has been reserved already if have, during reservation of VF(s), if any one of the VF reservation failed, the whole reservation will fail and rollback the reserved VFs. Also, reservation failure will raise exn.
So we can make assumtion that if any VF been reserved. all other VFs are supposed to be reserved.

Reply:
It's still checking per-VIF anyway, isn't it?

Reply:
Yes, it's still checking per-VIF, do you mean it's better once we found one VIF has SR-IOV reserved, than we can cancel the assertion?

Reply:
No, I don't think that's better. I thought from Ming's comment that he read your code comment that way. But I looks like you didn't mean that.

Reply:
Thanks, I had refined the comment above the function.